datacenterlight/tasks.py

```diff
@@ -94,33 +80,42 @@ def create_vm_task(self, vm_template_id, user, specs, template,
         if vm_id is None:
             raise Exception("Could not create VM")
 
-        # Create a Hosting Order
-        order = HostingOrder.create(
-            price=final_price,
-            vm_id=vm_id,
-            customer=customer,
-            billing_address=billing_address
+        # Update HostingOrder with the created vm_id
+        hosting_order = HostingOrder.objects.filter(id=order_id).first()
```

Comment:
How can we be sure it is the latest order? What if two orders have been placed in the meantime?

Reply:
Oh, id=order_id --- sorry, it's a bit early